src/array/cpu/cpu_id_hash_map.h

```diff
@@ -0,0 +1,151 @@
+/**
+ *  Copyright (c) 2019 by Contributors
+ * @file array/cpu/cpu_id_hash_map.h
+ * @brief Class about CPU id hash map
+ */
+
+#ifndef DGL_ARRAY_CPU_CPU_ID_HASH_MAP_H_
+#define DGL_ARRAY_CPU_CPU_ID_HASH_MAP_H_
+
+#include <dgl/aten/types.h>
+
+#include <vector>
+
+#ifdef _MSC_VER
+#include <intrin.h>
+#define COMPARE_AND_SWAP(ptr, oldval, newval, ret)                         \
```

Comment:
Do we need it in .h? what about just move to cc file if only you need this MACRO?

Reply:
If we move it to cc, we can make it a function instead of a MACRO.
```
#ifdef _MSC_VER
#include <intrin.h>
#else // _MSC_VER

template <typename IdType>
static IdType CompareAndSwap(IdType* ptr, IdType old, IdType new) {
#ifdef _MSC_VER
  if (sizeof(IdType) == 32) {
    return _InterlockedCompareExchange(...)
  } else if (sizeof(IdType) == 64) {
    ...
  } else {
    ...
  }
#elif __GNUC__  // _MSC_VER
  return __sync_val_compare_and_swap(...)
#else  // _MSC_VER
  #error "CompareAndSwap is not supported on this platform."
#endif  // _MSC_VER
}
```

Reply:
Good suggestion, changed to this style.